procrastinate/app.py

```diff
@@ -270,3 +269,30 @@ def health_check_runner(self) -> healthchecks.HealthCheckRunner:
     @property
     def admin(self) -> admin.Admin:
         return admin.Admin(connector=self.connector)
+
+    def open(self, pool: Optional[connector_module.Pool] = None) -> None:
+        self.connector.open(pool)
+
+    def close(self) -> None:
+        self.connector.close()
+
+    async def open_async(self, pool: Optional[connector_module.Pool] = None) -> None:
+        await self.connector.open_async(pool)
+
+    async def close_async(self) -> None:
+        await self.connector.close_async()
+
+    def __enter__(self, pool: Optional[connector_module.Pool] = None) -> None:
```

Comment:
`__enter__` normally takes a single argument, `self`. Why having an additional argument, `pool`, here?

Reply:
True, I noticed that this morning running unittests. This is pretty anoying, as we wanted our API to give the user the ability to specifiy the pool when entering the context (in `.enter` method for instance). We actually cannot do that. Maybe we should not allow the user specify the pool when entering the context. 

I will remove the user ability to provide the pool when entering context.

Reply:
> This is pretty anoying, as we wanted our API to give the user the ability to specifiy the pool when entering the context (in `.enter` method for instance).

Why did you want that?

When the pool is created externally it makes no sense to use a Procrastinate context manager. You'd use that instead:

```py
app = procrastinate.App(connector=procrastinate.AiopgConnector())
async with aiopg.create_pool() as pool:
    await app.open_async(pool)
    …
```

No?

Reply:
This is interesting, we should discuss it !